tests/unittests/HyphenTest.cpp

```diff
@@ -334,10 +333,15 @@ TEST(HyphenTest, network) {
 
   // Randomly shuffle the training data.
```

Comment:
At the moment this is the only unit test that uses it's own mt19937 random number generator. All other tests use nextRand().  Let's assume that all future tests would mimic this test. Do we want people to instantiate their own random number generator? Or maybe we want to encourage the tests to be more stable by changing the hyper parameters? Another alternative would be to adda reset function to the internal random number generator. 

What do you think?

Reply:
I think in general tests should be resilient to random gen, but if it's not possible (hard to achieve) explicit setting seed might be an option.

Reply:
Agreed about using stable tests. This network is clearly somewhat wobbly. I've increased the batch size further to 50, but there just isn't a lot of training data in this test case.

We could encourage tests to use the standard library, or we could provide our own pseudo-random number facility, either is fine by me. I do think the global state in `nextRand()` is problematic, though. Let's discuss more over in #1066.

I'm going to land this as is to unblock people from adding more JIT tests. When we reach a consensus on #1066, I can update this test to match.